web/logging/chargelog/ladelogexportexcel.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+function makedatetime($start,$f)
+{
+  //echo "$start;$f;";
+  
+   preg_match('/(.*)-(.*)/',$start,$xx);
+   $day=$xx[1];
+   $vonuhr=$xx[2];
+   preg_match('/(.*)-(.*)/',$f,$xx);
+   $bisuhr=$xx[2];
+   
+   echo "$day;$vonuhr - $bisuhr;";
+}
+
+
+
+ if( $_GET['do']=='export')
+ {
+   $dates='';
+   $year=0;
+   $fn='';
+
+   if( isset($_GET['date']) )
+   {
+      $dates=str_replace("-","",$_GET['date']);   
+      $fin="/var/www/html/openWB/web/logging/data/ladelog/$dates.csv";
+      $file=file($fin);
+      asort($file);
+      $fn='Ladelog_'.$dates.'.csv';
+   }
+   else if ( isset($_GET['year']) )
+   {
+      $year=str_replace("-","",$_GET['year']);
+   
+      $files = glob($_SERVER['DOCUMENT_ROOT'] . "/openWB/web/logging/data/ladelog/*.csv");
+      asort($files);
+      $rowClasses = "";
+      $file=[];
+      foreach ($files as $current) 
+      {
+        preg_match('/\/var\/www\/html\/openWB\/web\/logging\/data\/ladelog\/([0-9]{4})([0-9]{2})\.csv/',$current,$m);
+        if( $m[1] == $year )
+        { 
+          $onef = file($current);
+          asort($onef);
+          foreach($onef as $line)
+             if( trim($line) > '')
+               $file[]=$line;
+        }  
+      }
+      $fn='Ladelog_'.$year.'.csv';
+   }
+
+   
+   header('Content-Type: application/csv; charset=UTF-8');
+   header('Content-Disposition: attachment;filename="'.$fn.'";');
+
+   // Haben wir den Kilometerstand vom Ladestart im Ladelog ?   
```

Comment:
Bitte solche Anpassungen ohne weitere Verwendung in openWB rausnehmen.
Die Datei "soc1KM" gibt es nicht bei openWB.

Reply:
soc1Km kommt später hinzu.

Reply:
Dann pack die Zeilen doch auch später hinzu. Aktuell haben die hier nichts verloren.